### PR TITLE
SDIT-995 Add Non-association create sync service to nomis

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/visits/VisitService.kt [hunk 1]
```diff
@@ -196,7 +196,6 @@ class VisitService(
         startDateTime = updateVisitRequest.startDateTime,
         endDateTime = endDateTime,
         location = visit.location,
-        roomDescription = updateVisitRequest.room,
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/visits/VisitService.kt:199]: Did you mean for these to go back ?

Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/visits/VisitService.kt:199]: ah yes, i meant to mention these are intellij warnings about unused params etc

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/helper/AdjudicationOffenceTrigger.kt [hunk 2]
```diff
@@ -15,6 +15,6 @@ class AdjudicationOffenceTrigger : org.h2.api.Trigger {
     statement.setInt(1, Random.nextInt(999999))
     statement.setBigDecimal(2, incidentId)
     statement.setBigDecimal(3, chargeSequence)
-    val rows = statement.executeUpdate()
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/helper/AdjudicationOffenceTrigger.kt:18]: unused var warning

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/visits/VisitService.kt [hunk 3]
```diff
@@ -463,23 +460,6 @@ class VisitService(
...
-      .uppercase()
-      .take(240 - 12) // Full description can not exceed 240 characters (12 additional characters will be added to this description)
-      .removeSuffix("-")
-
-  private fun String.toInternalLocationCode(locationId: String) =
-    this.replace("VSIP", "VP")
-      .replace(locationId, "")
-      .replace("-", "")
-      .take(12) // Code does not need to be unique so just take the first 12 characters
-
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/visits/VisitService.kt:482]: unused funs

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**